src/lokijs.js

```diff
@@ -6283,11 +6298,14 @@
           }
 
           if (uic) {
-            Object.keys(this.constraints.unique).forEach(function (key) {
-              for (idx = 0; idx < len; idx++) {
-                doc = self.data[positions[idx]];
-                if (doc[key] !== null && doc[key] !== undefined) {
-                  self.constraints.unique[key].remove(doc[key]);
+            this.uniqueNames.forEach(function (key) {
```

Comment:
please note that with unique indices, we can be lazy and skip building the index until the first `by()` call - so if the index still hasn't been built, we don't need to update it on create/update/remove (this can actually be a nice performance boost on batch create). with `idIndex`, we often have to create it during create/update/remove, but not in all cases.